spec/datadog/tracing/integration_spec.rb

```diff
@@ -24,9 +24,31 @@
   shared_context 'agent-based test' do
```

Comment:
I'm overhauling this file in the Sampling-Propagation feature branch.
A few of the `shared_context`s have been backported here to ease testing and reduce merging conflicts.
They seem like overkill now, but will simplify and increase correctness of agent-based testing quite a bit.